plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -2,199 +2,240 @@
-    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_backgroundDrawable, -1);
-    addBackgroundLayer(ContextCompat.getDrawable(context, drawableResId));
-    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_bearingDrawable, -1);
-    addBearingLayer(ContextCompat.getDrawable(context, drawableResId));
-    drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_navigationDrawable, -1);
-    addNavigationLayer(ContextCompat.getDrawable(context, drawableResId));
-    float accuracyAlpha = typedArray.getFloat(R.styleable.LocationLayer_accuracyAlpha, 0.15f);
-    if (accuracyAlpha < 0 || accuracyAlpha > 1) {
-      throw new RuntimeException("Location layer accuracy alpha value must be between 0.0 and 1.0.");
-    }
-    int accuracyColor = typedArray.getColor(R.styleable.LocationLayer_accuracyColor,
-      ContextCompat.getColor(context, R.color.mapbox_plugin_location_layer_blue));
-    addAccuracyLayer(accuracyAlpha, accuracyColor);
-
-    typedArray.recycle();
-  }
-
-  void setLayerVisibility(boolean visible) {
-    List<Layer> layers = mapboxMap.getLayers();
-    String id;
-    for (Layer layer : layers) {
-      id = layer.getId();
-      if (layerIds.contains(layer.getId())) {
-        // TODO use sourceLayer filter instead
-        if (id.equals(LocationLayerConstants.LOCATION_ACCURACY_LAYER)
-          || id.equals(LocationLayerConstants.LOCATION_BEARING_LAYER)
-          || id.equals(LocationLayerConstants.LOCATION_LAYER)
-          || id.equals(LocationLayerConstants.LOCATION_BACKGROUND_LAYER)
-          || id.equals(LocationLayerConstants.LOCATION_NAVIGATION_LAYER)) {
-          layer.setProperties(visibility(visible ? Property.VISIBLE : Property.NONE));
-        }
-      }
-    }
-  }
+    try {
+      int drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_foregroundDrawable, -1);
+      styleForeground(ContextCompat.getDrawable(context, drawableResId));
 
-  void setCompassBearing(float compassBearing) {
-    Layer bearingLayer = mapboxMap.getLayer(LocationLayerConstants.LOCATION_BEARING_LAYER);
-    if (bearingLayer != null) {
-      bearingLayer.setProperties(PropertyFactory.iconRotate(compassBearing));
+      drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_backgroundDrawable, -1);
+      styleBackground(ContextCompat.getDrawable(context, drawableResId));
+
+      drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_bearingDrawable, -1);
+      styleBearing(ContextCompat.getDrawable(context, drawableResId));
+
+      drawableResId = typedArray.getResourceId(R.styleable.LocationLayer_navigationDrawable, -1);
+      styleNavigation(ContextCompat.getDrawable(context, drawableResId));
+
+      float accuracyAlpha = typedArray.getFloat(R.styleable.LocationLayer_accuracyAlpha, 0.15f);
+      if (accuracyAlpha < 0 || accuracyAlpha > 1) {
+        throw new RuntimeException("Location layer accuracy alpha value must be between 0.0 and 1.0.");
```

Comment:
can be more specific here with `IllegalArgumentException`

Reply:
I did not really change anything to that code block, was copied from previous implementation. Seeing that you are already addressing it in https://github.com/mapbox/mapbox-plugins-android/pull/186/files#diff-6b76d34c67041b321a2ce642ab12a240R75, I think we can ignore this for now?